### PR TITLE
Move ringpop-go to glide

### PR DESCRIPTION
**Pull request by Motiejus**
* Works on a local OSX machine with a fresh `$GOPATH`.
* Works on travis-ci.
* **Does not** work on a fresh Docker `golang:1.6` instance due to possibly different thrift version (not verified). However, the `dev` branch doesn't work too, so it's a separate problem.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,3 +29,7 @@ script:
 
 after_success:
   - test/update-coveralls
+
+cache:
```
Comment (dansimau) [inline .travis.yml]: 👍

Comment (Motiejus) [inline .travis.yml]: I didn't try too hard with this, I am sure we can do even more. I just picked the lowest hanging fruit here.

**Comment by thanodnl** [comment]
godep is still mentioned in the `README.md` files of the examples eg: https://github.com/uber/ringpop-go/blob/motiejus-glide/examples/ping-json/README.md

I think it is good to review those and update them to reflect reality. From my tests it works to checkout the repo, run `make setup` and after that `cd` to the example directory and make a build via `go build`. `go run` works as well.

**Comment by mennopruijssers** [comment]
- Is the `--debug` when running `glide` a leftover from some testing? (https://github.com/uber/ringpop-go/blob/motiejus-glide/Makefile#L51)
- It might be "better" for readability to use `glide novendor` instead of it's `nv` shorthand. (e.g. https://github.com/uber/ringpop-go/blob/motiejus-glide/Makefile#L65)

Some versions in `glide.lock` are different than the ones in `Godeps.json`. Is this on purpose or by accident? Also two new packages (`github.com/uber-go/atomic` and `golang.org/x/sys`)? 

Package | Old | New
------- | --- | ---
github.com/apache/thrift | 5bc8b5a3a5da507b6f87436ca629be664496a69f | 39a09ac5e49481d39dd1bcb6757ffe182e3df20a
github.com/cactus/go-statsd-client | 8947bd0d9e04e8c27671c33753340b51b835efdc | 91c326c3f7bd20f0226d3d1c289dd9f8ce28d33d
github.com/dgryski/go-farm | fc41e106ee0e4394f69a56046e5c5adbaf29d912 | f94e93e5a2b3a579af0f5367c84b54cb9aa00074
github.com/pmezard/go-difflib | e8554b8641db39598be7f6342874b958f12ae1d4 | 792786c7400a136282c1664665ae0a8db921c6c2
github.com/rcrowley/go-metrics | dee209f2455f101a5e4e593dea94872d2c62d85d | eeba7bd0dd01ace6e690fa833b3f22aaec29af43
github.com/Sirupsen/logrus | 82c8de1af23b25ba5a468d801f29077f4f483bce | f3cfb454f4c209e6668c95216c4744b8fddb2356
github.com/stretchr/testify | e3a8ff8ce36581f87a15341206f205b1da467059 | 8d64eb7173c7753d6419fd4a9caf057398611364
github.com/uber/tchannel-go | 6ea263463a946c89e6565a12110aa424175155c3 | 6d2feae147d9f2c5142b335128e568b874adfd56
golang.org/x/net | 5aa7325eaa14d7ed4b520f40d58adf2834c8de01 | c4c3ea71919de159c9e246d7be66deb7f0a39a58

**Comment by mennopruijssers** [comment]
Also there are still references to godep:
```
$ git grep godep
examples/ping-json/README.md:godep go build
examples/ping-thrift-gen/README.md:godep go build
examples/ping-thrift/README.md:godep go build
```

**Comment by Motiejus** [comment]
@mennopruijssers @thanodnl I updated everything regarding to your comments.

**Comment by Motiejus** [comment]
Sorry about the last commit. It was a merge commit with conflicts, so I did the following:

* `git rm -fr Godeps`
* `open glide.lock`, update version of statsd.

I should have split the merge commit into two: merge and conflict resolution.

**Comment by Motiejus** [comment]
I am also adding tests for `examples/` directories to never have the build instructions to fail, but seems like one of the READMEs is incorrect, so the test fails (not in travis yet). Here is the branch, I expect to have it in travis-ci very shortly: https://github.com/uber/ringpop-go/tree/motiejus-glide-examples

File: examples/ping-thrift/README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ thrift-gen --generateThrift --outputDir gen-go --inputFile ping.thrift
 
 Build the example binary:
 ```bash
-godep go build
+go build
```
Comment (mennopruijssers) [inline examples/ping-thrift/README.md]: Don't we need a `glide install` first?

**Comment by mennopruijssers** [comment]
Small question about the readme's but besides it: looks good!

**Comment by Motiejus** [comment]
`examples/` directory needs updated build instructions. I tried to use `vendor/` with glide, but that fails:

```
cp ../../glide.{yaml,lock} .
glide install --cache
<...>
motiejus ~/.gopath/src/github.com/uber/ringpop-go/examples/ping-json $ go build
# github.com/uber/ringpop-go/examples/ping-json
./main.go:85: cannot use "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".JSON (type "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".Format) as type "github.com/uber/tchannel-go".Format in argument to w.ringpop.HandleOrForward
./main.go:113: cannot use ch (type *"github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".Channel) as type shared.TChannel in argument to ringpop.Channel:
        *"github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".Channel does not implement shared.TChannel (wrong type for GetSubChannel method)
                have GetSubChannel(string, ..."github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".SubChannelOption) *"github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber/tchannel-go".SubChannel
                want GetSubChannel(string, ..."github.com/uber/tchannel-go".SubChannelOption) *"github.com/uber/tchannel-go".SubChannel
./main.go:115: cannot use "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber-common/bark".NewLoggerFromLogrus(logger) (type "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber-common/bark".Logger) as type "github.com/uber-common/bark".Logger in argument to ringpop.Logger:
        "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber-common/bark".Logger does not implement "github.com/uber-common/bark".Logger (wrong type for Fields method)
                have Fields() "github.com/uber/ringpop-go/examples/ping-json/vendor/github.com/uber-common/bark".Fields
                want Fields() "github.com/uber-common/bark".Fields
```

Looks like this is related: https://groups.google.com/forum/#!topic/golang-nuts/liKQ6gtU4hk

**Comment by Motiejus** [comment]
I will rework this guide to initialize glide. This will be useful for newcommers too.

**Comment by Motiejus** [comment]
@mennopruijssers can you check again? I updated the `examples/` directory.

**Comment by dansimau** [comment]
I think there is an issue here if the user has `GO15VENDOREXPERIMENT=0`.

Here's what I got:

```
dan-mbp-work:ringpop-go dan$ make test
[ERROR] To use Glide, you must set GO15VENDOREXPERIMENT=1
[ERROR] To use Glide, you must set GO15VENDOREXPERIMENT=1
go generate 
test/go-test-prettify 
ok  	_/tmp/ringpop-go	0.103s

# tests: 52
# passed: 52

[...]
```

**Comment by dansimau** [comment]
Also, it looks as though we're now trying to run tests for the Go code in ringpop-common:

```
dan-mbp-work:ringpop-go dan$ make
go generate ./discovery/... ./events/... ./examples/... ./forward/... ./hashring/... ./logging/... ./replica/... ./router/... ./scripts/... ./shared/... ./swim/... ./test/... ./util/... .
Generating mock for: TChanRemoteService
test/go-test-prettify ./discovery/... ./events/... ./examples/... ./forward/... ./hashring/... ./logging/... ./replica/... ./router/... ./scripts/... ./shared/... ./swim/... ./test/... ./util/... .
test/ringpop-common/cluster-test/test-orchestrator/calc.go:34:2: cannot find package "github.com/pkg/errors" in any of:
	/tmp/go/src/github.com/uber/ringpop-go/vendor/github.com/pkg/errors (vendor tree)
	/usr/local/Cellar/go/1.6.2/libexec/src/github.com/pkg/errors (from $GOROOT)
	/tmp/go/src/github.com/pkg/errors (from $GOPATH)
test/ringpop-common/cluster-test/test-orchestrator/test_yaml_parser.go:32:2: cannot find package "gopkg.in/yaml.v2" in any of:
	/tmp/go/src/github.com/uber/ringpop-go/vendor/gopkg.in/yaml.v2 (vendor tree)
	/usr/local/Cellar/go/1.6.2/libexec/src/gopkg.in/yaml.v2 (from $GOROOT)
	/tmp/go/src/gopkg.in/yaml.v2 (from $GOPATH)

# tests: 0

make: *** [test-unit] Error 1
dan-mbp-work:ringpop-go dan$ 
```

This is probably an unrelated issue due to Go code being landed in ringpop-common. Feel free to create a separate issue for it.

**Comment by mennopruijssers** [comment]
LGTM

**Comment by Motiejus** [comment]
@dansimau I updated the pull request to override `$GO15VENDOREXPERIMENT` in the Makefile and tested locally. Could you do another try?

**Comment by dansimau** [comment]
LGTM

**Pull request merged**